src/App.js

```diff
@@ -1,36 +1,188 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { library } from "@fortawesome/fontawesome-svg-core";
+import { fas } from "@fortawesome/free-solid-svg-icons";
+library.add(fas);
 
 function App() {
-  const location = useLocation();
-  const background = location.state && location.state.background;
+
+  const [show, setShow] = useState(false);
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  const [addNewTaskTitle, setaddNewTaskTitle] = useState("");
+  const [taskDescription, settaskDescription] = useState("");
+  const [dueDate, setDueDate] = useState(new Date());
+
+  const postData = () => {
+    axios
+      .post(`https://64ef1ed7219b3e2873c3f9ad.mockapi.io/todoData`, {
+        addNewTaskTitle,
+        taskDescription,
+        dueDate,
+      })
+      .then(() => {
+        getData();
+      });
+  };
+
+  //// from read
+
+  const [APIData, setAPIData] = useState([]);
+  useEffect(() => {
+    axios
+      .get(`https://64ef1ed7219b3e2873c3f9ad.mockapi.io/todoData`)
+      .then((response) => {
+        setAPIData(response.data);
+      });
+  }, []);
+
+  const setData = (data) => {
+    console.log(data);
+    let { id, addNewTaskTitle, taskDescription, dueDate } = data;
+    localStorage.setItem("ID", id);
+    localStorage.setItem("Your Task", addNewTaskTitle);
+    localStorage.setItem("Due Date", dueDate);
+    localStorage.setItem("To-Do", taskDescription);
+  };
+
+  const onDelete = (id) => {
+    axios
+      .delete(`https://64ef1ed7219b3e2873c3f9ad.mockapi.io/todoData/${id}`)
+      .then(() => {
+        getData();
+      });
+  };
+  const getData = () => {
+    axios
+      .get(`https://64ef1ed7219b3e2873c3f9ad.mockapi.io/todoData`)
+      .then((getData) => {
+        setAPIData(getData.data);
+      });
+  };
+
 
   return (
-      <div className="App">
-        <div className='todo-menu'>
-            <p>
-              To-DO List
-            </p>
-            <Routes location={background || location}>
-              <Route path="/" element={<AddTaskButton />}>
-              <Route path='/read' element={<Read />} />
-                <Route path="modal" element={<Create />} />
-              </Route>
-              {/* <Route path='/read' element={<Read />} /> */}
-              <Route path='/update' element={<Update />} />
-              </Routes>
-                {background && (
-              <Routes>
-                <Route path="modal" element={<Create />} />
-              </Routes>
-      )}
-          </div>
+    <div className="App">
+      <div className="todo-menu">
+        <p>To-DO List</p>
+        <div>
+          <Button className="button-add-tasks" onClick={handleShow}>
+            Add New Task
+            <FontAwesomeIcon icon="fa-solid fa-plus" className="plus-solid" />
+          </Button>
+
+          <Modal show={show} onHide={handleClose}>
+            <Modal.Header closeButton>
```

Comment:
Please move this part to a new component. We'll work on this at another time.

Reply:
Done, I moved Modal to the ModalForm component.